src/Raven.Server/Documents/PeriodicBackup/Azure/RavenAzureClient.cs

```diff
@@ -114,11 +125,11 @@ private string GetUrl(string containerName, bool isTest = false)
         private async Task PutBlockApi(string key, Stream stream, Dictionary<string, string> metadata)
         {
             if (stream.Length > TotalBlocksSizeLimitInBytes)
-                throw new InvalidOperationException($"Can't upload more than 195GB to Azure, current upload size: {stream.Length/1024/1024/1024}GB");
+                throw new InvalidOperationException($"Can't upload more than 4.75TB to Azure, current upload size: {stream.Length/1024D/1024/1024/1024}TB");
 
             var threads = ProcessorInfo.ProcessorCount/2 + 1;
-            //max size of in memory queue is: 100MB
-            var queue = new BlockingCollection<ByteArrayWithBlockId>(Math.Min(25, threads*2));
+            //max size of in memory queue is: 500MB
+            var queue = new BlockingCollection<ByteArrayWithBlockId>(Math.Min(5, threads*2));
```

Comment:
This is a lot, why are we holding so much in memory? Why are we using threads for this?